### PR TITLE
Chore/3.0.0 rc.2

### LINKED ISSUE #211: Add support for chapters to CldVideoPlayer

### LINKED ISSUE #213: Hydration missmatch for CLD Video Player
Run playground and see the error in the console

image

### LINKED ISSUE #214: Bug when running docs locally
1. Install deps
2. Run dev

image

### LINKED ISSUE #216: [Feature] Use getVideoPlayerOptions to support CldVideoPlayer
### Is your feature request related to a problem? Please describe.

getVideoPlayer options was released as a function available in the @cloudinary-util/url-loader package allowing us to maintain the logic to support the component in a shared module.

Using it is as simple as passing the options / props directly to the function, returning the options that will then be passed to the player.

```
const playerOptions = getVideoPlayerOptions(props, config);
videoPlayer(videoRef.current, playerOptions);
```

An additional argument is the `config` object, which is shaped as ConfigOptions, currently being used in the library.

Next Cloudinary: https://github.com/cloudinary-community/next-cloudinary/pull/478

Cloudinary Util: https://github.com/cloudinary-community/cloudinary-util/pull/157

### LINKED ISSUE #217: CldOgImage does not render the image in Twitter
Due to the fact that by default the value for the TwitterTitle property is `'  '` it does not render correctly the OG image on twitter.

### LINKED ISSUE #218: fix the issue with `resize` and `widthResize` props in CldImage
5.0.0 which removed widthResize in favor of baseResize https://github.com/cloudinary-community/cloudinary-util/pull/143
then removed base* i nfavor of the crop parameter: https://github.com/cloudinary-community/cloudinary-util/pull/144

### PR DESCRIPTION
**Pull request by Baroshem**
Closes #217 
Closes #213 
Closes #214
Closes #218 
Closes #211 
Closes #216

### CODE REVIEW

File: docs/content/2.components/CldOgImage.md [hunk 1]
```diff
@@ -27,9 +27,21 @@ The resulting HTML will be applied to the Head of the document:
...
 
+Keep in mind that if you want to render an OG image in Twitter, you need to:
+
+1. Have a `currentRoute.meta?.title` set to some value
+2. Pass a prop to the component like following:
+
+```html
+<CldOgImage
+  src="cld-sample-2"
+  twitter-title="My Awesome Twitter Title"
```
Comment (colbyfayock) [inline docs/content/2.components/CldOgImage.md:43]: should this be `twitterTitle` or is that a Nuxt convention to turn it to kebab case?

File: docs/content/2.components/CldOgImage.md [hunk 2]
```diff
@@ -27,9 +27,21 @@ The resulting HTML will be applied to the Head of the document:
...
 <meta content="627" property="og:image:height">
-<meta content=" " property="twitter:title" />
+<meta content="My awesome Twitter Title" property="twitter:title" />
 <meta content="summary_large_image" property="twitter:card">
 <meta content="https://res.cloudinary.com/nuxt-cloudinary/image/upload/c_fill,w_2400,h_1254,g_center/c_scale,w_1200/f_webp/q_auto/cld-sample-2?_a=BBDAACAD0" property="twitter:image">
 ```
 
+Keep in mind that if you want to render an OG image in Twitter, you need to:
+
+1. Have a `currentRoute.meta?.title` set to some value
```
Comment (colbyfayock) [inline docs/content/2.components/CldOgImage.md:37]: just to confirm is this available during SSR/SSG? as it won't work if it's loaded clientside

File: docs/content/2.components/CldOgImage.md (hunk 1 above)
Comment (colbyfayock) [reply docs/content/2.components/CldOgImage.md:43]: guessing thats the case based on other examples im seeing

File: src/runtime/components/CldOgImage.vue [hunk 3]
```diff
@@ -125,7 +117,10 @@ const { url: twitterImageUrl } = useCldImageUrl({
 } as ConstructUrlProps)
 
 const computedTwitterTitle = computed(
-  () => props.twitterTitle || (currentRoute.value.meta?.title as string) || ' ',
+  () =>
+    props.twitterTitle
+    || (currentRoute.value.meta?.title as string)
+    || 'nuxt-cloudinary-og-image',
```
Comment (colbyfayock) [inline src/runtime/components/CldOgImage.vue:123]: thoughts on something like "Learn More" instead? it's overly generic, but at least it's potentially somewhat useful if they dont  set it

File: docs/content/2.components/CldOgImage.md (hunk 2 above)
Comment (Baroshem) [reply docs/content/2.components/CldOgImage.md:37]: Yes it is available there :)

File: docs/content/2.components/CldOgImage.md (hunk 1 above)
Comment (Baroshem) [reply docs/content/2.components/CldOgImage.md:43]: Vue uses different naming for registering and using props.

So in this case the prop name is actually `twitterTitle` but how we use it in Vue component is like `twitter-title`.

**Pull request merged**